dan_layer/engine_types/src/substate.rs

```diff
@@ -746,19 +753,19 @@ mod tests {
                 .as_resource_address()
                 .unwrap();
             SubstateId::from_str(
-                "resource_7cbfe29101c24924b1b6ccefbfff98986d648622272ae24f7585dab5ffffffff nft_str:SpecialNft",
+                "resource_7cbfe29101c24924b1b6ccefbfff98986d648622272ae24f7585dab5ffffffff_nft_str:SpecialNft",
```

Comment:
Issue #1080 specifies the format as
`nft_{resource}_{type}_{nft_id}`

Reply:
Ok, I wasn't sure since the functions being updated handle multiple types of ID strings. I will make it uniform across the board with this format.